lib/runner.js

```diff
@@ -322,6 +322,8 @@ Runner.prototype.hook = function (name, fn) {
             hook.pending = true;
           }
         } else {
+          // set the current test to the next one to run so the error message is correct
+          hook.ctx.currentTest = suite.tests[0];
```

Comment:
what happens if this is an "after" hook?

Reply:
Sorry for the late response, I missed your comment.
I have added a test case for the after hook, it seems to be working as expected.
I also had to make an additional change as a test for the bail option was failing.
Please have another look :)